### DISCUSSION TITLE
Can InheritConfig cope with nesting?

### DISCUSSION THREAD

**Discussion by andrewducker** [renovatebot/renovate#29967] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab - latest version of Renovate

### Please tell us more about your question or problem

We have multiple levels of nesting of our projects.  The owner of one of the high-level groups wants to apply a setting to all of the projects within all of the subgroups.
I thought that the inheritconfig setting would do that, but looking at the logs it only seems to look in the direct parent of the repo.

So I thought that if the repo is at `Org\FrontEnd\Java\Libraries\OneOfOurLibraries` (for example) that I could place the config to be inherited in the `Org\FrontEnd\renovate-config\` and it would check all the way up the tree until it found it.

Is that something that Renovate supports?  If not, is there any way to apply some config to every repo in a given subtree?

**Comment by rarkins** [comment, marked as answer]
Renovate does not support nested/recursive inheritConfig hunting.

If you put it at the very top Group then you could set `inheritConfigRepoName` to `{{topLevelOrg}}//renovate-config`.

The type of recursive hunting you describe would take up a few extra API requests per repo btw (which could consume the API limit)

**Reply by andrewducker** [reply]
We're running our own Gitlab instance, so that's not a concern for us. 

But thank you, we'll have a think about what we can do to work around this. 

(Our actual design is such that the level will vary. We may have to go and update a few dozen repos to use "extends")
